XCUITests/WebAuthenticationTest.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  WebAuthenticationTest.swift
+//  Client
+//
+//  Created by mozilla on 12/23/16.
+//  Copyright © 2016 Mozilla. All rights reserved.
+//
+
+import XCTest
+import GCDWebServers
+
+class WebAuthenticationTest: BaseTestCase {
+private var webRoot: String!
+        
+    override func setUp() {
+       
+        super.setUp()
+        dismissFirstRunUI()
+        webRoot = SimplePageServer.start()
+        continueAfterFailure = false
+    }
+    
+    override func tearDown() {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+        super.tearDown()
+    }
+
+    private func loadAuthPage() {
+        let app = XCUIApplication()
+
+        app.textFields["url"].tap()
+        app.textFields["address"].typeText("\(webRoot)/auth.html\n")
```

Comment:
You can try setting an explicit port like we do here. https://github.com/mozilla-mobile/firefox-ios/blob/ff1e0b9a37099c67027618ce58cc5365a1e79d5d/Client/Application/WebServer.swift#L35